## tratar os dados $\longrightarrow$ modelo $\longrightarrow$ o que o modelo quer dizer pra mim com os dados $\longrightarrow$ selecionar os sensores mais importantes $\longrightarrow$ acho que construir serie temportal para os sensores mais relevantes (importantes).



### csv 06 = parece temperatura
### csv 31 = parece ser pressao kpc

### alguns dados estão com NaN

### fazer um df trandando os dados e jogado no modelo. 




In [192]:
import pandas as pd 
import glob

caminho_csvs = "C:/Users/Thierry/Documents/IC/ps/new_data"


sensores = glob.glob(caminho_csvs + "/TAG_IALL_PS_*.csv") 
sensor_taget = pd.read_csv(caminho_csvs + '/target_iALL_PS.csv')

dado_frame = sensor_taget.copy()
for i in sensores:
    sensor = pd.read_csv(i) #== i = 0 = [0] = 
    dado_frame = pd.merge(dado_frame, sensor, on='timestamp', how='left')



sensor_coluna = [coluna for coluna in dado_frame.columns if coluna.startswith('TAG')]
 
print("dimensão df:")
print(dado_frame.shape)

print("tipo:")
print(dado_frame.dtypes) #object pq combinação de da ta ano/mes/dia

#float tipo do dado 4 bytes na memoria 

dimensão df:
(220320, 54)
tipo:
timestamp          object
target_iALL_PS     object
TAG_iALL_PS_00    float64
TAG_iALL_PS_01    float64
TAG_iALL_PS_02    float64
TAG_iALL_PS_03    float64
TAG_iALL_PS_04    float64
TAG_iALL_PS_05    float64
TAG_iALL_PS_06    float64
TAG_iALL_PS_07    float64
TAG_iALL_PS_08    float64
TAG_iALL_PS_09    float64
TAG_iALL_PS_10    float64
TAG_iALL_PS_11    float64
TAG_iALL_PS_12    float64
TAG_iALL_PS_13    float64
TAG_iALL_PS_14    float64
TAG_iALL_PS_15    float64
TAG_iALL_PS_16    float64
TAG_iALL_PS_17    float64
TAG_iALL_PS_18    float64
TAG_iALL_PS_19    float64
TAG_iALL_PS_20    float64
TAG_iALL_PS_21    float64
TAG_iALL_PS_22    float64
TAG_iALL_PS_23    float64
TAG_iALL_PS_24    float64
TAG_iALL_PS_25    float64
TAG_iALL_PS_26    float64
TAG_iALL_PS_27    float64
TAG_iALL_PS_28    float64
TAG_iALL_PS_29    float64
TAG_iALL_PS_30    float64
TAG_iALL_PS_31    float64
TAG_iALL_PS_32    float64
TAG_iALL_PS_33    float64
TAG_iALL_PS_34    float64
TAG_iA

In [193]:
print(sensores)

['C:/Users/Thierry/Documents/IC/ps/new_data\\TAG_iALL_PS_00.csv', 'C:/Users/Thierry/Documents/IC/ps/new_data\\TAG_iALL_PS_01.csv', 'C:/Users/Thierry/Documents/IC/ps/new_data\\TAG_iALL_PS_02.csv', 'C:/Users/Thierry/Documents/IC/ps/new_data\\TAG_iALL_PS_03.csv', 'C:/Users/Thierry/Documents/IC/ps/new_data\\TAG_iALL_PS_04.csv', 'C:/Users/Thierry/Documents/IC/ps/new_data\\TAG_iALL_PS_05.csv', 'C:/Users/Thierry/Documents/IC/ps/new_data\\TAG_iALL_PS_06.csv', 'C:/Users/Thierry/Documents/IC/ps/new_data\\TAG_iALL_PS_07.csv', 'C:/Users/Thierry/Documents/IC/ps/new_data\\TAG_iALL_PS_08.csv', 'C:/Users/Thierry/Documents/IC/ps/new_data\\TAG_iALL_PS_09.csv', 'C:/Users/Thierry/Documents/IC/ps/new_data\\TAG_iALL_PS_10.csv', 'C:/Users/Thierry/Documents/IC/ps/new_data\\TAG_iALL_PS_11.csv', 'C:/Users/Thierry/Documents/IC/ps/new_data\\TAG_iALL_PS_12.csv', 'C:/Users/Thierry/Documents/IC/ps/new_data\\TAG_iALL_PS_13.csv', 'C:/Users/Thierry/Documents/IC/ps/new_data\\TAG_iALL_PS_14.csv', 'C:/Users/Thierry/Docume

In [194]:
print(dado_frame.head())

             timestamp target_iALL_PS  TAG_iALL_PS_00  TAG_iALL_PS_01  \
0  2018-04-01 00:00:00         NORMAL        4.548754       90.886874   
1  2018-04-01 00:01:00         NORMAL        7.887998       56.555373   
2  2018-04-01 00:02:00         NORMAL        4.975919      182.086958   
3  2018-04-01 00:03:00         NORMAL        6.304142       58.417235   
4  2018-04-01 00:04:00         NORMAL        1.671733      108.946809   

   TAG_iALL_PS_02  TAG_iALL_PS_03  TAG_iALL_PS_04  TAG_iALL_PS_05  \
0       58.698896       89.301134     1011.733181       97.284889   
1       80.802810      120.898222     1358.466600      202.583688   
2       87.273632        9.914782     1056.489015      147.098428   
3       75.059520       64.167463     1619.924847      307.722320   
4       94.910470       14.551922      591.648283       -7.684779   

   TAG_iALL_PS_06  TAG_iALL_PS_07  ...  TAG_iALL_PS_42  TAG_iALL_PS_43  \
0       33.248746      -10.390227  ...      101.547023      154.291476  

In [195]:
print("Quantidade de NaN:")
print(dado_frame.isnull().sum())

Quantidade de NaN:
timestamp              0
target_iALL_PS         0
TAG_iALL_PS_00     10208
TAG_iALL_PS_01       369
TAG_iALL_PS_02        19
TAG_iALL_PS_03        19
TAG_iALL_PS_04        19
TAG_iALL_PS_05        19
TAG_iALL_PS_06      4798
TAG_iALL_PS_07      5451
TAG_iALL_PS_08      5107
TAG_iALL_PS_09      4595
TAG_iALL_PS_10        19
TAG_iALL_PS_11        19
TAG_iALL_PS_12        19
TAG_iALL_PS_13        19
TAG_iALL_PS_14        21
TAG_iALL_PS_15    220320
TAG_iALL_PS_16        31
TAG_iALL_PS_17        46
TAG_iALL_PS_18        46
TAG_iALL_PS_19        16
TAG_iALL_PS_20        16
TAG_iALL_PS_21        16
TAG_iALL_PS_22        41
TAG_iALL_PS_23        16
TAG_iALL_PS_24        16
TAG_iALL_PS_25        36
TAG_iALL_PS_26        20
TAG_iALL_PS_27        16
TAG_iALL_PS_28        16
TAG_iALL_PS_29        72
TAG_iALL_PS_30       261
TAG_iALL_PS_31        16
TAG_iALL_PS_32        68
TAG_iALL_PS_33        16
TAG_iALL_PS_34        16
TAG_iALL_PS_35        16
TAG_iALL_PS_36        16
TAG_iA

In [196]:
print(dado_frame.count())
#então que dizer que o dataset 15 nao tem nada. O sensor não funciona..... 

timestamp         220320
target_iALL_PS    220320
TAG_iALL_PS_00    210112
TAG_iALL_PS_01    219951
TAG_iALL_PS_02    220301
TAG_iALL_PS_03    220301
TAG_iALL_PS_04    220301
TAG_iALL_PS_05    220301
TAG_iALL_PS_06    215522
TAG_iALL_PS_07    214869
TAG_iALL_PS_08    215213
TAG_iALL_PS_09    215725
TAG_iALL_PS_10    220301
TAG_iALL_PS_11    220301
TAG_iALL_PS_12    220301
TAG_iALL_PS_13    220301
TAG_iALL_PS_14    220299
TAG_iALL_PS_15         0
TAG_iALL_PS_16    220289
TAG_iALL_PS_17    220274
TAG_iALL_PS_18    220274
TAG_iALL_PS_19    220304
TAG_iALL_PS_20    220304
TAG_iALL_PS_21    220304
TAG_iALL_PS_22    220279
TAG_iALL_PS_23    220304
TAG_iALL_PS_24    220304
TAG_iALL_PS_25    220284
TAG_iALL_PS_26    220300
TAG_iALL_PS_27    220304
TAG_iALL_PS_28    220304
TAG_iALL_PS_29    220248
TAG_iALL_PS_30    220059
TAG_iALL_PS_31    220304
TAG_iALL_PS_32    220252
TAG_iALL_PS_33    220304
TAG_iALL_PS_34    220304
TAG_iALL_PS_35    220304
TAG_iALL_PS_36    220304
TAG_iALL_PS_37    220304


In [197]:
print(dado_frame.shape)

(220320, 54)


In [198]:
#verificar o dataset 00.


csv_escolha = 'TAG_iALL_PS_00'
nan_ds0 = dado_frame[csv_escolha].isna()
nanmax= dado_frame[nan_ds0]['timestamp'].max()
nanmin = dado_frame[nan_ds0]['timestamp'].min()
nanmaxtil = dado_frame[~nan_ds0]['timestamp'].max()
nanmintil = dado_frame[~nan_ds0]['timestamp'].min()
print(nanmax)
print(nanmin)
print(nanmaxtil)
print(nanmintil)

2018-08-31 18:50:00
2018-04-12 22:00:00
2018-08-31 23:59:00
2018-04-01 00:00:00


In [199]:
print(dado_frame.shape)

(220320, 54)


In [200]:
#como o dataset 15 esta vazio, vou descartar ele do data frame 100% vazio 

dado_frame = dado_frame.drop(columns=['TAG_iALL_PS_15'])

print(dado_frame.shape[0])
print(dado_frame.dropna().shape[0])
print(dado_frame.shape[0] - dado_frame.dropna().shape[0])

220320
119103
101217


In [201]:
print(dado_frame.shape)

(220320, 53)


In [202]:
# #pegar uma amostra 1 a cada 5 min 

# dado_frame = dado_frame.iloc[::5].reset_index(drop=True)

# print(dado_frame.shape[0]) #-> qunatidade das linhas das amostras 
# print(dado_frame.dropna().shape[0]) #->quantidade das linhas das amostras se os nan

In [203]:
print(dado_frame.shape)

(220320, 53)


In [204]:
print(dado_frame['target_iALL_PS'].value_counts())

target_iALL_PS
NORMAL     205836
ANORMAL     14484
Name: count, dtype: int64


In [205]:
print(dado_frame['target_iALL_PS'].value_counts(normalize=True)*100)

target_iALL_PS
NORMAL     93.425926
ANORMAL     6.574074
Name: proportion, dtype: float64


In [206]:
print(dado_frame.shape)

(220320, 53)


### veja que incialmente após criar um data frame com so csv's eu saio de um data frame de dimensão (220320, 54) para um data frame final  de dimensão (8813, 53) 

In [207]:
print(dado_frame)

                  timestamp target_iALL_PS  TAG_iALL_PS_00  TAG_iALL_PS_01  \
0       2018-04-01 00:00:00         NORMAL        4.548754       90.886874   
1       2018-04-01 00:01:00         NORMAL        7.887998       56.555373   
2       2018-04-01 00:02:00         NORMAL        4.975919      182.086958   
3       2018-04-01 00:03:00         NORMAL        6.304142       58.417235   
4       2018-04-01 00:04:00         NORMAL        1.671733      108.946809   
...                     ...            ...             ...             ...   
220315  2018-08-31 23:55:00         NORMAL        6.066461      128.157145   
220316  2018-08-31 23:56:00         NORMAL        4.057784      -31.186502   
220317  2018-08-31 23:57:00         NORMAL        6.751912       80.433446   
220318  2018-08-31 23:58:00         NORMAL        2.084228      192.076427   
220319  2018-08-31 23:59:00         NORMAL        4.558682       16.293402   

        TAG_iALL_PS_02  TAG_iALL_PS_03  TAG_iALL_PS_04  TAG_iAL

In [208]:
dado_frame = dado_frame
%store dado_frame


Stored 'dado_frame' (DataFrame)
